source/SH3/system/glprogram.cpp

```diff
@@ -55,6 +55,50 @@ GLint sh3_glprogram::Load(const std::string& name)
     glDeleteShader(fragShader);
 }
 
+GLint sh3_glprogram::Load(const std::string& name, std::vector<std::string> const &attribs)
+{
+    GLint status = 0;
+
+    programID = glCreateProgram();
+
+    // Generate our vertex and fragment shader
+    vertShader = (GLuint)Compile(GL_VERTEX_SHADER);
+    fragShader = (GLuint)Compile(GL_FRAGMENT_SHADER);
+
+    glAttachShader(programID, vertShader);
+    glAttachShader(programID, fragShader);
+
+    // Bind any attributes before we link the program
+    for(int i = 0; i < attribs.size(); i++)
+    {
+        glBindAttribLocation(programID, i, reinterpret_cast<const GLchar*>(attribs[i].c_str()));
```

Comment:
You can have any attribute name you want I believe. I think attributes are sort of like executable symbols like `.rodata`, `.text` and `.data` (though I'm not 100% certain on this), which is why they need to be specified pre link. This is the only way the CPU can actually interface with the GPU to send it data... I still don't have a clear picture though..

I'll add some error checking for all the errors it can throw, I'm sure it probably will across different graphics cards, God knows the differences nVidia and AMD have across their drivers....